shared/actions/profile/proofs.js

```diff
@@ -145,9 +146,32 @@ function* addProof(state, action) {
     }
   })
 
+  const watchForNavUp = yield Saga._fork(function*() {
+    // TODO remove this when nav1 is removed
+    // nav2 should use navigationOptions.gesturesEnabled = false
+    // to disable sidestepping custom behavior on back
+    while (true) {
+      yield Saga.take(RouteTreeGen.navigateUp)
+      const state = yield Saga.selectState()
+      const path = getPath(state.routeTree.routeState)
+      if (
+        ![
```

Comment:
navigating up is probably always invalid here; the point of this is just to make sure we don't leave a hanging RPC and the user unable to make new proofs.